app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user&.authenticate params[:session][:password]
```

Comment:
&. để làm gì e ?

Reply:
&. là Safe Navigation Operator cho phép chương trình chạy một cách an toàn mà k cần lo rằng có object nil hay không ạ